core/src/fetch_stage.rs

```diff
@@ -85,40 +90,26 @@ impl FetchStage {
     }
 
     fn handle_forwarded_packets(
-        recvr: &PacketBatchReceiver,
-        sendr: &PacketBatchSender,
+        recvr: &BoundedPacketBatchReceiver,
+        sendr: &BoundedPacketBatchSender,
         poh_recorder: &Arc<Mutex<PohRecorder>>,
     ) -> Result<()> {
         let mark_forwarded = |packet: &mut Packet| {
             packet.meta.flags |= PacketFlags::FORWARDED;
         };
 
-        let mut packet_batch = recvr.recv()?;
-        let mut num_packets = packet_batch.packets.len();
-        packet_batch.packets.iter_mut().for_each(mark_forwarded);
-        let mut packet_batches = vec![packet_batch];
-        while let Ok(mut packet_batch) = recvr.try_recv() {
-            packet_batch.packets.iter_mut().for_each(mark_forwarded);
-            num_packets += packet_batch.packets.len();
-            packet_batches.push(packet_batch);
-            // Read at most 1K transactions in a loop
-            if num_packets > 1024 {
-                break;
-            }
-        }
+        let (mut packet_batches, num_packets) = recvr.recv(RECV_MAX_PACKETS)?;
```

Comment:
change to 1024 to preserve previous behavior

Reply:
Changed back to 1024